pandas/core/dtypes/common.py

```diff
@@ -220,6 +220,10 @@ def is_categorical(arr):
     return isinstance(arr, ABCCategorical) or is_categorical_dtype(arr)
 
 
+def is_range(arr):
```

Comment:
```python
In [3]: pd.Series([1,2,3]).to_sparse().dtype
Out[3]: dtype('int64')
```
... still,
```python
In [4]: _concat.get_dtype_kinds([pd.Series([1,2,3]).to_sparse()])
Out[4]: {'sparse'}
```
... so the type returned by ``get_dtype_kinds`` is already _not_ the dtype, and not even the ``dtype.kind``.

But anyway, I don't particularly care about changing ``get_dtype_kinds``. We just need a method which can tell us whether two indexes can be natively concatenated: this is currently ``get_dtype_kinds``, but I can write a new one if you prefer.